haystack/utils/hf.py

```diff
@@ -275,8 +278,31 @@ def convert_message_to_hf_format(message: ChatMessage) -> Dict[str, Any]:
         # HF does not provide a way to communicate errors in tool invocations, so we ignore the error field
         return hf_msg
 
-    if text_contents:
+    # Handle multimodal content (text + images) for user messages
```

Comment:
I would prefer to use a for-loop like in Bedrock (https://github.com/deepset-ai/haystack-core-integrations/pull/2114)
- preserves the order of the parts
- (hopefully) it simplifies the implementation